From eff69d4461e346ca6455d08b8234f348691b8157 Mon Sep 17 00:00:00 2001
From: Phil Edworthy <phil.edworthy@renesas.com>
Date: Mon, 9 Jan 2017 10:14:30 +0000
Subject: [PATCH] mtd: nand: eva: Only check used OOB bytes are zero

Signed-off-by: Phil Edworthy <phil.edworthy@renesas.com>
---
 drivers/mtd/nand/evatronix_nand.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/nand/evatronix_nand.c b/drivers/mtd/nand/evatronix_nand.c
index 38d8ec5..f3617f2 100644
--- a/drivers/mtd/nand/evatronix_nand.c
+++ b/drivers/mtd/nand/evatronix_nand.c
@@ -1204,8 +1204,10 @@ static int check_erased_page(struct mtd_info *mtd, uint8_t *buf, int len)
 		 NFC_READ_OOB);
 
 	/* We go for the simple approach and accept eccstrength zero bits */
-	if (count_zero_bits(nfc_info->dma.buf, mtd->oobsize, eccstrength) >
-	    eccstrength)
+	/* We only check the BBM and ECC bytes, the rest may be file system */
+	if (count_zero_bits(nfc_info->dma.buf,
+		(chip->ecc.bytes * chip->ecc.steps) + ECC_OFFSET,
+		eccstrength) > eccstrength)
 		return 0;
 
 	MTD_TRACE("%s: Page is erased.%s\n", __func__,
-- 
2.7.4

