From 17f7e2943a3809bd89a5d26dc0e881f309b1495e Mon Sep 17 00:00:00 2001
From: Phil Edworthy <phil.edworthy@renesas.com>
Date: Wed, 8 Feb 2017 10:50:02 +0000
Subject: [PATCH] mtd: nand: eva: Fix section mismatch

Signed-off-by: Phil Edworthy <phil.edworthy@renesas.com>
---
 drivers/mtd/nand/evatronix_nand.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/nand/evatronix_nand.c b/drivers/mtd/nand/evatronix_nand.c
index f3617f2..12287bc 100644
--- a/drivers/mtd/nand/evatronix_nand.c
+++ b/drivers/mtd/nand/evatronix_nand.c
@@ -1813,7 +1813,7 @@ static const struct mtd_ooblayout_ops nfc_ooblayout_ops = {
 };
 
 /* Per-NAND-chip initialization. */
-static __init
+static
 struct mtd_info *nfc_flash_probe(struct platform_device *pdev,
 				 unsigned int bank_no)
 {
@@ -1965,7 +1965,7 @@ struct mtd_info *nfc_flash_probe(struct platform_device *pdev,
 }
 
 /* Main probe function. Called to probe and set up device. */
-static int __init nfc_probe(struct platform_device *pdev)
+static int nfc_probe(struct platform_device *pdev)
 {
 	struct device *dev = &pdev->dev;
 	struct mtd_info *main_mtd;
-- 
2.7.4

